src/Metatables.cs

```diff
@@ -1037,38 +1037,47 @@ bool TrySetMember(LuaState luaState, ProxyType targetType, object target, Bindin
                 member = members[0];
                 SetMemberCache(targetType, fieldName, member);
             }
-
-            if (member.MemberType == MemberTypes.Field)
+            else if (member is LuaNativeFunction)
             {
-                var field = (FieldInfo)member;
-                object val = _translator.GetAsType(luaState, 3, field.FieldType);
-
-                try
-                {
-                    field.SetValue(target, val);
-                }
-                catch (Exception e)
-                {
-                    ThrowError(luaState, e);
-                }
-
+                SetMemberCache(targetType, fieldName, member);
                 return true;
             }
-            if (member.MemberType == MemberTypes.Property)
+            else if (member is MemberInfo)
```

Comment:
Maybe we can drop the `else` if we early return the first `if` as well.

Reply:
Ah, I missed that return. Will fix.